packages/babel-generator/src/printer.js

```diff
@@ -58,7 +58,7 @@ export default class Printer extends Buffer {
     if (needsParens) this.push(")");
 
     // end
-    this.map.mark(node, "end");
+    if (parent) this.map.mark(parent);
```

Comment:
I don't really understand what's happening here? Why do we add more mappings to the parent?